appium/options/base_options.py

```diff
@@ -0,0 +1,34 @@
+from typing import Any, TypeVar, Generic, Type, Callable, Optional
+
+from appium.options.common.supports_capabilities import SupportsCapabilities
+
+RT = TypeVar("RT")
+VT = TypeVar("VT")
+C = TypeVar('C', bound='SupportsCapabilities')
+
+class OptionsDescriptor(Generic[RT, VT]):
+    """Generic Descriptor class which calls get_capability and set_capability:
+    without transforming the values
+
+    If transformation method is passed, get_capability and set_capability:
+    method will be called with transformed value
+    """
+    def __init__(self, name: str, 
+                 tget: Optional[Callable[[Any], RT]] = None, 
+                 tset: Optional[Callable[[VT], Any]] = None) -> None:
+        self.name = name
+        self.tget = tget
+        self.tset = tset
+
+    def __get__(self, obj: C, cls: Type[C]) -> RT:
+        value = obj.get_capability(self.name)
+        if self.tget:
+            # transform the value and then return
+            return self.tget(obj, value)
```

Comment:
the signature of this function does not match to its type definition. Please run mypy as mentioned above